src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,235 @@
+package duke.ui;
+
+import duke.task.Task;
+import duke.tasklist.TaskList;
+import duke.exception.DukeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Ui class handles all user interactions for the Duke application. It is responsible
+ * for displaying messages to the user, including greetings, task operations feedback, and errors.
+ */
+public class Ui {
+    
+    /**
+     * Prints a message to the user indicating a task has been added, and displays the current
+     * total number of tasks.
+     *
+     * @param task The task that has been added.
+     * @param list The task list to which the task was added.
+     */
+    public void echo(Task task, TaskList list) {
+        String echo = "Got it. I've added this task:\n" + "  " + task.toString() + "\n"
+            + "Now you have " + (list.size() + 1) + " tasks in the list"
+            + "\n___________________________________" ;
+        System.out.println(echo);
+        list.add(task);
+    }
+    
+    /**
+     * Prints a greeting message to the user when the application starts.
+     */
+    public void greeting() {
+        String greeting = "___________________________________\n"
+            + "Hello! I'm Jinni\n"
+            + "What can I do for you?\n"
+            + "___________________________________" ;
+        System.out.println(greeting);
+    }
+    
+    /**
+     * Lists all tasks currently in the task list, displaying each task's status and description.
+     *
+     * @param list The task list containing the tasks to be listed.
+     */
+    public void listing(TaskList list) {
+        System.out.println("Here are the tasks in your list\n");
+        if (list.size() == 0) {
+            System.out.println("\n___________________________________");
+        } else {
+            int num = 1;
+            for (Task t : list.getList()) {
```

Comment:
Nice that you initialized the variables where they are declared, and in the smallest scope possible.